src/hotspot/share/oops/klass.inline.hpp

```diff
@@ -76,4 +77,70 @@ inline ByteSize Klass::vtable_start_offset() {
   return in_ByteSize(InstanceKlass::header_size() * wordSize);
 }
 
+// subtype check: true if is_subclass_of, or if k is interface and receiver implements it
+inline bool Klass::is_subtype_of(Klass* k) const {
+  guarantee(secondary_supers() != nullptr, "must be");
```

Comment:
Minor point: considering libjvm contains hundreds of copies, does it make sense to turn it into an assert instead? For example, on AArch64 the check costs 36 bytes [1] in product build.  

[1]
```
libjvm.dylib[0x1306d4] <+28>:  ldr    x9, [x8, #0x28]
libjvm.dylib[0x1306d8] <+32>:  cbz    x9, 0x1307dc              ; <+292> [inlined] Klass::is_subtype_of(Klass*) const at klass.inline.hpp:82:3
...
libjvm.dylib[0x1307dc] <+292>: adrp   x0, 3200
libjvm.dylib[0x1307e0] <+296>: add    x0, x0, #0x663            ; "open/src/hotspot/share/oops/klass.inline.hpp"
libjvm.dylib[0x1307e4] <+300>: adrp   x2, 3200
libjvm.dylib[0x1307e8] <+304>: add    x2, x2, #0x690            ; "guarantee(secondary_supers() != nullptr) failed"
libjvm.dylib[0x1307ec] <+308>: adrp   x3, 3200
libjvm.dylib[0x1307f0] <+312>: add    x3, x3, #0x6c0            ; "must be"
libjvm.dylib[0x1307f4] <+316>: mov    w1, #0x52
libjvm.dylib[0x1307f8] <+320>: bl     0x54f870                  ; report_vm_error at debug.cpp:181
```